knowledge_graph/labelling.py

```diff
@@ -1,310 +1,443 @@
+        user_object = self.get_user(username)
+
+        try:
+            user_object.add_to_workspace(workspace_object)
+            logger.info("Added user '%s' to workspace '%s'", username, workspace_name)
+        except ValueError:
+            raise
+        except Exception as e:
+            raise ValueError(
+                f"Failed to add user '{username}' to workspace '{workspace_name}'"
+            ) from e
+
+    def remove_user_from_workspace(
+        self, username: str, workspace: Optional[str] = None
+    ):
+        """Remove a user from a workspace"""
+        workspace_name = workspace or self.default_workspace
+        logger.info("Removing user '%s' from workspace '%s'", username, workspace_name)
+
+        workspace_object = self.get_workspace(workspace_name)
+        user_object = self.get_user(username)
+
+        try:
+            user_object.remove_from_workspace(workspace_object)
+            logger.info(
+                "Removed user '%s' from workspace '%s'", username, workspace_name
+            )
+        except ValueError:
+            raise
+        except Exception as e:
+            raise ValueError(
+                f"Failed to remove user '{username}' from workspace '{workspace_name}'"
+            ) from e
+
+    def add_labelled_passages(
         self,
-        concept: Concept,
-        workspace: str = "knowledge-graph",
-        min_timestamp: Optional[datetime] = None,
-        max_timestamp: Optional[datetime] = None,
-    ) -> list[LabelledPassage]:
+        labelled_passages: list[LabelledPassage],
+        wikibase_id: WikibaseID | str,
+        workspace: Optional[str] = None,
+    ) -> Dataset:
         """
-        Get the labelled passages from Argilla for a given concept.
-
-        :param Concept concept: The concept to get the labelled passages for
-        :param Optional[datetime] min_timestamp: Only get annotations made after this timestamp (inclusive), defaults to None
-        :param Optional[datetime] max_timestamp: Only get annotations made before this timestamp (inclusive), defaults to None
-        :raises ValueError: If no dataset matching the concept ID was found in Argilla
-        :raises ValueError: If no datasets were found in Argilla, you may need to be granted access to the workspace(s)
-        :return list[LabelledPassage]: A list of LabelledPassage objects
+        Add labelled passages to an existing dataset in Argilla.
+
+        Args:
+            labelled_passages: List of LabelledPassage objects to add. Note that only
+                the text and the metadata of each passage will be uploaded - spans
+                attached to the labelled passages will be ignored.
+            wikibase_id: Wikibase ID of the dataset to add passages to.
+            workspace: Name of the workspace to add passages to. If not provided, the
+                session's default_workspace will be used.
+
+        Returns:
+            The updated dataset.
```

Comment:
i've said this before, but I think it'd be good if we could stick to one docstring format everywhere – and I think the one with colons (sphinx?) is more used.